### ISSUE TITLE
The plugin.(*GRPCProvider).ConfigureProvider request was cancelled

### ISSUE THREAD

**Issue by tlfzhylj** [hashicorp/terraform-provider-azurerm#24822]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.7.3

### AzureRM Provider Version

3.90.0

### Affected Resource(s)/Data Source(s)

provider "azurerm" {}

### Terraform Configuration Files

```hcl
provider "azurerm" {
  features {}
  subscription_id            = "b8d098d9-c5db-4afd-9138-7af0eabf9ec1" # not a real sub id
  alias                      = "example1"
}

provider "azurerm" {
  features {}
  subscription_id            = "2bf6928a-cdf1-4956-8bda-aaa7f080d162" # not a real sub id
  alias                      = "example2"
}

provider "azurerm" {
  features {}
  subscription_id            = "cf043e6b-01cb-4e51-9cf9-6c2f262eb4f5" # not a real sub id
  alias                      = "example3"
}
```

### Debug Output/Panic Output

```shell
##[error]╷
│ Error: Request cancelled
│ 
│   with provider["registry.terraform.io/hashicorp/azurerm"].example1,
│   on environments.mimir.tf line 5, in provider "azurerm":
│    5: provider "azurerm" {
│ 
│ The plugin.(*GRPCProvider).ConfigureProvider request was cancelled.
╵
╷
│ Error: Request cancelled
│ 
│   with provider["registry.terraform.io/hashicorp/azurerm"].example2,
│   on environments.mimir.tf line 21, in provider "azurerm":
│   21: provider "azurerm" {
│ 
│ The plugin.(*GRPCProvider).ConfigureProvider request was cancelled.
╵
```

### Actual Behaviour

My theory is that this happens because I have several provider call to the azurerm provider, with different aliases to different Azure subscriptions. It looks like it works ok with 6 or 7, but when adding number 8 it fails every time. It also fails sometimes when running a plan with 7 provider calls to azurerm.

### Steps to Reproduce

terraform plan

### Important Factoids

Happens when running in my Azure pipeline. Can't run locally because of storage account with tfstate is locked down with private endpoints.

### References

https://github.com/jason-johnson/azure-pipelines-tasks-terraform/issues/412
Before updating to azurerm provider version 3.90 (from 3.50) I didn't get any good error message. Then I got this.

**Comment by tlfzhylj** [comment]
I experienced the same error in another pipeline which had an older Terraform version, but then Terraform just gave me a `null` in the exit code, which resulted in the error messages in my Azure pipeline:
`##[error]Terraform command 'plan' failed with exit code 'null'.`
`##[error]TypeError: Cannot read property 'toString' of null`

To get the error message The plugin.(*GRPCProvider).ConfigureProvider request was cancelled I had to update Terraform version from 1.5 to 1.7, and the azurerm provider from 3.50 to 3.90.

**Comment by tlfzhylj** [comment]
I figured out the actual problem:

In my Terraform configuration I configure the azurerm provider several times to go against several Azure subscriptions. If I configured the azurerm provider six times against six different Azure subscriptions or more I got the error.

When configuring the provider so many times it looks like it uses more memory than on a normal run with one configured azurerm provider. My self-hosted Azure DevOps agent had 1 GB of memory. When increasing the memory to 2 GB everything works as expected.

So, the problem was actually that the Azure DevOps Agent ran out of memory.

I will probably hit the same error if configuring around 12 azurerm providers, after increasing the memory to 2 GB.
